### ISSUE TITLE
Some nodes in flannel have network failure problems

### ISSUE THREAD

**Issue by rainbow211** [coreos/flannel#1182]
**Expected Behavior**
Flannel`v0.10.0` assigns a unique subnet to each node and runs well

**Current Behavior**    
   Recently, when using flannel in a large-scale cluster of enterprise clouds, it was found that two nodes were assigned subnet information.Is the same network segment address, after repeated troubleshooting, we found that when the flanger is uninstalled, it will not be cleared.In addition to the local file **/run/flannel/subnet.env**. When the flannel is started, it will load the local text directly.The subnet in /run/flannel/subnet.env, and does not dynamically get the subnet, so the direct consequence of this is that it may lead to the flannel device of several nodes and the network allocated by the **docker0** network segment.A conflict has occurred, causing serious network communication problems. 

**Possible Solution**
       My suggestion is to go dynamically when the flannel starts.Get the subnet and synchronize to the local file **/run/flannel/subnet.env**.Read directly in the local file when the flannel is running.

**Comment by rajatchopra** [comment]
We need to clean up the subnet.env file on cleanup. PRs welcome.

**Comment by rainbow211** [comment]
> We need to clean up the subnet.env file on cleanup. PRs welcome.

I am very happy to hear from you, I still have two questions.
① subnet.env file is created with the flannel running, then why do we need to help the flannel to clean up the environment when uninstalling,instead of flannel? Or the latest version of flannel support to clean up subnet.env?  
② After I uninstall the flannel, check the ip route will also find the route left by the flannel, this still needs manual cleaning?
In short,Wouldn't it be better if the flannel could do everything for the user, including cleaning up everything it brought?
Looking forward to your reply!
